### PR TITLE
Give your unique weapons some backstory!

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Cobby [Idea stolen from Shaps]
add: For objects that you could previously rename with a pen, you can now edit their description as well.
/:cl:

Lore-Lovers : *heavy breathing*

Setup a pastebin now so you can copypaste your novel-backstory of your .38 without having to write it everytime.

### CODE REVIEW

**Comment by calzilla1** [comment]
Oh Jesus christ. Now i can make Nigger burger with the description of made with 100% free range dindu when the chef puts out custom burgurs w/o nameing em

**Comment by Shadowlight213** [comment]
That's WGW.
One day while Andy....

Never 4get when another admin and I discovered that revolver naming had no character limit.

**Comment by FantasticFwoosh** [comment]
Making Mead will never be the same again.

_According to all known laws_
_of aviation,_
  
_there is no way a bee_
_should be able to fly._

_Its wings are too small to get_
_its fat little body off the ground._

_The bee, of course, flies anyway..._

**Comment by Kromgar** [comment]
Are we bay12 now?

**Comment by ExcessiveUseOfCobblestone** [comment]
This only affects objects that have `unique_rename = 1`, so for the most part only the person writing the description will ever actually view the description.

This also has a character limit equivalent to the amount you can put in a Communications Console broadcast message.

**Comment by Supermichael777** [comment]
you underestimate them.

**Comment by ExcessiveUseOfCobblestone** [comment]
Even if they DO have a portion of WGW or Bee Movie in the object's description, who cares? No one is going to see it unless they examine the object while it's out and about.

**Comment by TechnoAlchemisto** [comment]
IS there a character limit for this?

**Comment by Shadowlight213** [comment]
Yes. It's relatively high though.

**Comment by ExcessiveUseOfCobblestone** [comment]
I wasn't sure how indepth people planned on making their kinetic accelerator or .38 backstory. I guess I should start low and raise it as needed depending on if people like the feature instead of starting high and lowering it depending on how many people meme with it.

**Comment by ExcessiveUseOfCobblestone** [comment]
It now has a 100 character limit, enough to give it some very short or flashy description..

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -21,7 +25,13 @@
 		M << "You name the weapon [input]. Say hello to your new friend."
 		return
 
+/obj/item/weapon/proc/redesc_weapon(mob/M)
+	var/input = stripped_input(M,"Describe your object here", ,"", 100)
 
+	if(src && input && !M.stat && in_range(M,src) && !M.restrained() && M.canmove)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: if(!qdeleted(src) && canUseTopic(src, be_close = TRUE))

   update rename_weapon to use the same check if does not already

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -5,12 +5,16 @@
...
 	..()
 	if(unique_rename)
-		user << "<span class='notice'>Use a pen on it to rename it.</span>"
+		user << "<span class='notice'>Use a pen on it to rename it or change its description.</span>"
 
 /obj/item/weapon/attackby(obj/item/I, mob/user, params)
 	if(unique_rename)
 		if(istype(I, /obj/item/weapon/pen))
-			rename_weapon(user)
+			var/penchoice = alert("What would you like to edit?", "Rename or change description?", "Rename", "Change description", "Cancel")
```
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: This is a sleep so you need

    if(!qdeleted(src) && canUseTopic(src, be_close = TRUE))

**Review by optimumtact — CHANGES_REQUESTED** [review]
needs some changes

**Review by Iamgoofball — APPROVED** [review]

**Review by optimumtact — APPROVED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
Fully tested and functioning 👌

**Pull request merged**